cmake/Modules/FindMKL.cmake

```diff
@@ -212,7 +238,8 @@ MACRO(CHECK_ALL_LIBRARIES LIBRARIES OPENMP_TYPE OPENMP_LIBRARY _name _list _flag
         # Separately handling compiled TBB
         SET(_found_tbb TRUE)
       ELSE()
-        FIND_LIBRARY(${_prefix}_${_library}_LIBRARY NAMES ${_library})
+        SET(lib_names ${_library} ${_library}_dll)
+        FIND_LIBRARY(${_prefix}_${_library}_LIBRARY NAMES ${lib_names})
```

Comment:
My only concern is that introducing the shared libs here may break some of our release builds. Could you please do that in a separate PR?

Reply:
> 
> 
> My only concern is that introducing the shared libs here may break some of our release builds. Could you please do that in a separate PR?

OK